File: Treachery-iOS/Treachery-iOSTests/PlayerColorsTests.swift
import Testing
@testable import Treachery_iOS

struct PlayerColorsTests {

    @Test func paletteHas10Colors() {
        #expect(PlayerColors.palette.count == 10)
    }

    @Test func allColorsHaveNames() {
        for color in PlayerColors.palette {
            #expect(!color.name.isEmpty)
        }
    }

    @Test func allColorsHaveHexValues() {
        for color in PlayerColors.palette {
            #expect(!color.hex.isEmpty)
            #expect(color.hex.count == 6 || color.hex.count == 7) // With or without #
        }
    }

    @Test func allColorsHaveUniqueHex() {
        let hexes = PlayerColors.palette.map(\.hex)
        #expect(Set(hexes).count == hexes.count)
    }

    @Test func allColorsHaveUniqueNames() {
        let names = PlayerColors.palette.map(\.name)
        #expect(Set(names).count == names.count)
    }
}

File: TreacheryAndroid/app/src/test/java/com/solomon/treachery/data/PlayerColorsTests.kt
package com.solomon.treachery.data

import com.solomon.treachery.ui.util.PlayerColors
import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.Test

/**
 * Tests for the PlayerColors palette.
 * Mirrors iOS PlayerColorsTests.
 */
class PlayerColorsTests {

    @Test
    fun `palette has 10 colors`() {
        assertEquals(10, PlayerColors.palette.size)
    }

    @Test
    fun `all colors have non-empty names`() {
        PlayerColors.palette.forEach { color ->
            assertTrue(color.name.isNotBlank(), "Color should have a name")
        }
    }

    @Test
    fun `all colors have valid hex values`() {
        PlayerColors.palette.forEach { color ->
            assertTrue(color.hex.isNotBlank(), "Color '${color.name}' should have a hex value")
            // Hex should be 6 or 7 chars (with or without #)
            val cleaned = color.hex.removePrefix("#")
            assertEquals(6, cleaned.length, "Color '${color.name}' hex should be 6 digits, got '${color.hex}'")
        }
    }

    @Test
    fun `all hex values are unique`() {
        val hexValues = PlayerColors.palette.map { it.hex }
        assertEquals(hexValues.size, hexValues.toSet().size, "Duplicate hex values found")
    }

    @Test
    fun `all names are unique`() {
        val names = PlayerColors.palette.map { it.name }
        assertEquals(names.size, names.toSet().size, "Duplicate color names found")
    }
}
